webapp/src/views/projects/translations/context/services/useEditService.ts

```diff
@@ -0,0 +1,233 @@
+    const key = translations.fixedTranslations?.find(
+      (k) => k.keyId === position?.keyId
+    );
+    if (key) {
+      return (
+        (position?.language
+          ? key.translations[position.language]?.text
+          : key.keyName) || ''
+      );
+    }
+  };
+
+  const mutateTranslationKey = async (payload: SetEdit) => {
+    if (payload.value !== getEditOldValue()) {
+      await putKey.mutateAsync({
+        path: { projectId: project.id, id: payload.keyId },
+        content: {
+          'application/json': {
+            name: payload.value,
+          },
+        },
+      });
+    }
+  };
+
+  const mutateTranslation = async (
+    payload: SetEdit,
+    languagesToReturn?: string[]
+  ) => {
+    const { language, value, keyId } = payload;
+    const keyName = getTranslationKeyName(keyId);
+
+    const newVal =
+      payload.value !== getEditOldValue()
+        ? await putTranslation.mutateAsync({
+            path: { projectId: project.id },
+            content: {
+              'application/json': {
+                key: keyName,
+                translations: {
+                  [language!]: value,
+                },
+                languagesToReturn,
+              },
+            },
+          })
+        : null;
+    return newVal;
+  };
+
+  const setPositionAndFocus = (pos: Edit | undefined) => {
+    // make it async if someone is stealing focus
+    setTimeout(() => {
+      // focus cell when closing editor
+      if (pos === undefined && position) {
+        const newPosition = {
+          keyId: position.keyId,
+          language: position.language,
+        };
+        viewRefs.focusCell(newPosition);
+      }
+      setPosition(pos);
+    });
+  };
+
+  const setEdit = (newPosition: Edit | undefined) => {
+    if (position?.changed) {
+      setPositionAndFocus({ ...position, mode: 'editor' });
+      confirmation({
+        title: t('translations_unsaved_changes_confirmation_title'),
+        message: t('translations_unsaved_changes_confirmation'),
+        cancelButtonText: t('back_to_editing'),
+        confirmButtonText: t('translations_cell_save'),
+        onConfirm: () => {
+          changeField({
+            onSuccess() {
+              setPositionAndFocus(newPosition);
+            },
+          });
+        },
+      });
+    } else {
+      setPositionAndFocus(newPosition);
+    }
+  };
+
+  const changeField = async (data: ChangeValue) => {
+    if (!position) {
+      return;
+    }
+    const { keyId, language, value } = position;
+    if (!language && !value) {
+      // key can't be empty
+      return messaging.error(t('global_empty_value'));
+    }
+    try {
+      if (language) {
+        // update translation
+        await mutateTranslation(
+          {
+            ...data,
+            value: value as string,
+            keyId,
+            language,
+          },
+          translations.selectedLanguages
+        ).then((data) => {
```

Comment:
When there is already try/await, shouldn't we follow this pattern without using ".then" block?

Reply:
Yep, it does make sense